accounts-db/src/accounts_db.rs

```diff
@@ -3284,7 +3285,14 @@ impl AccountsDb {
                                                 {
                                                     assert!(slot <= &max_clean_root_inclusive);
                                                 }
-                                                purges_old_accounts.push(*candidate);
+                                                if slot_list.len() > 1 {
+                                                    // no need to purge old accounts if there is only 1 slot in the slot list
+                                                    purges_old_accounts.push(*candidate);
```

Comment:
the key observation here is that when `purges_old_accounts` is used, it only does useful work when the slot_list.len() > 1:
https://github.com/anza-xyz/agave/blob/d7353b038be622ac265b90536dda071977a9d926/accounts-db/src/accounts_db.rs#L2680